### PR TITLE
hw-6(2)

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,31 @@ public Integer getLength() {
...
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return width == null && rectangle.width == null
+                || width != null && width.equals(rectangle.width)
+                && length == null && rectangle.length == null
+                || length != null && length.equals(rectangle.length)
+                && color == null && rectangle.color == null
+                || color != null && color.equals(rectangle.color);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: The logical-AND operator ( && ) has higher precedence than the logical-OR operator ( || )
add parenthesis to this statement to ensure the flow is correct

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (ghost) [reply src/main/java/core/basesyntax/Rectangle.java]: done

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,31 @@ public Integer getLength() {
...
+                && (length == null && rectangle.length == null
+                || length != null && length.equals(rectangle.length))
+                && (color == null && rectangle.color == null
+                || color != null && color.equals(rectangle.color));
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 24;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: is it a prime number?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (ghost) [reply src/main/java/core/basesyntax/Rectangle.java]: understood and corrected :)

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**